streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -274,10 +275,26 @@ private void shutdown() {
         log.info("Stream thread shutdown complete [" + this.getName() + "]");
     }
 
+    /**
+     * Compute the latency based on the current marked timestamp,
+     * and update the marked timestamp with the current system timestamp.
+     *
+     * @return latency
+     */
+    private long computeLatency() {
```

Comment:
I think the start time should be passed in here (see previous comment about `currTimeMs` and then this method just becomes: `Math.max(time.milliseconds - startTimeMs, 0)`